backend/cn/solver.mli

```diff
@@ -29,24 +40,28 @@ val provable :
   LogicalConstraints.t ->
   [> `True | `False ]
 
-
+(* Ask the solver for the model that it found in a call to [provable] *)
 val model :
   unit ->
   model_with_q
 
-
-
+(* Ask the solver to evaluate a CN term in the context of a
+   model.  (Might return None in case we ask for the value of a "don't
+   care" value in the (minimal) model.) *)
+(* TODO: BCP: I don't understand how this could ever be called -- how do we get a
+   model to pass it??? *)
 val eval :
-  Global.t ->
+  Global.t ->     (* TODO: BCP: IIUC Christopher thinks this is not needed? *)
```

Comment:
Here and in other places in the solver module, I think, the `global` argument can be removed: the solver is initialised with a `global` typing context at some point earlier on, and then this is expected not to change any longer.